acceptance: tests should try to kill leftover containers from failed previous invocations.

I ran `make acceptance` against sha `093e1e3` and encountered the following panic:

```
+ go test -tags acceptance ./acceptance -run . -timeout 10m -v -nodes 3
I160608 09:58:36.104754 rand.go:76  Random seed: -7777799812709100292
=== RUN   TestAdminLossOfQuorum
I160608 09:58:36.156787 acceptance/cluster/localcluster.go:658  stopping
--- FAIL: TestAdminLossOfQuorum (0.05s)
panic: Error response from daemon: network cockroachdb_acceptance has active endpoints [recovered]
        panic: Error response from daemon: network cockroachdb_acceptance has active endpoints [recovered]
        panic: Error response from daemon: network cockroachdb_acceptance has active endpoints

goroutine 9 [running]:
panic(0x5217c20, 0xc8202e8030)
        /usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
testing.tRunner.func1(0xc820084510)
        /usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:467 +0x192
panic(0x5217c20, 0xc8202e8030)
        /usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:443 +0x4e9
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).stopOnPanic(0xc82012e1c0)
        /Users/jordan/repo/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:238 +0x9b
panic(0x5217c20, 0xc8202e8030)
        /usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:443 +0x4e9
github.com/cockroachdb/cockroach/acceptance/cluster.maybePanic(0xb740028, 0xc8202e8030)
        /Users/jordan/repo/src/github.com/cockroachdb/cockroach/acceptance/cluster/docker.go:166 +0x4b
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).createNetwork(0xc82012e1c0)
        /Users/jordan/repo/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:268 +0x2ef
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).Start(0xc82012e1c0)
        /Users/jordan/repo/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:588 +0xc2
github.com/cockroachdb/cockroach/acceptance.StartCluster(0xc820084510, 0xc82017ab60, 0x9, 0xc82017c4e0, 0x1, 0x1, 0x12a05f200, 0x0, 0x0)
        /Users/jordan/repo/src/github.com/cockroachdb/cockroach/acceptance/util_test.go:184 +0x3d1
github.com/cockroachdb/cockroach/acceptance.runTestOnConfigs.func1(0xc820084510, 0xc82013bec8, 0x580b240)
        /Users/jordan/repo/src/github.com/cockroachdb/cockroach/acceptance/util_test.go:157 +0x34
github.com/cockroachdb/cockroach/acceptance.runTestOnConfigs(0xc820084510, 0x580b240)
        /Users/jordan/repo/src/github.com/cockroachdb/cockroach/acceptance/util_test.go:160 +0x247
github.com/cockroachdb/cockroach/acceptance.TestAdminLossOfQuorum(0xc820084510)
        /Users/jordan/repo/src/github.com/cockroachdb/cockroach/acceptance/admin_test.go:34 +0x2d
testing.tRunner(0xc820084510, 0x6056d80)
        /usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:473 +0x98
created by testing.RunTests
        /usr/local/Cellar/go/1.6.2/libexec/src/testing/testing.go:582 +0x892
exit status 2
FAIL    github.com/cockroachdb/cockroach/acceptance     0.100s
```

--- a/acceptance/cluster/localcluster.go
+++ b/acceptance/cluster/localcluster.go
@@ -261,12 +261,18 @@ func (l *LocalCluster) panicOnStop() {
 func (l *LocalCluster) createNetwork() {
 	l.panicOnStop()
 
-	nets, err := l.client.NetworkList(context.Background(), types.NetworkListOptions{})
-	maybePanic(err)
-	for _, net := range nets {
-		if net.Name == networkName {
-			maybePanic(l.client.NetworkRemove(context.Background(), net.ID))
+	net, err := l.client.NetworkInspect(context.Background(), networkName)
+	if err == nil {
+		// We need to destroy the network and any running containers inside of it.
+		for containerID := range net.Containers {
+			// This call could fail if the container terminated on its own after we call
+			// NetworkInspect, but the likelihood of this seems low. If this line creates
+			// a lot of panics we should do more careful error checking.
+			maybePanic(l.client.ContainerKill(context.Background(), containerID, "9"))
 		}
+		maybePanic(l.client.NetworkRemove(context.Background(), networkName))
+	} else if !client.IsErrNotFound(err) {
+		panic(err)
 	}
 
 	resp, err := l.client.NetworkCreate(context.Background(), networkName, types.NetworkCreate{

